pkg/util/ecs/metadata/v3or4/client.go

```diff
@@ -22,20 +22,22 @@ import (
 )
 
 const (
-	// DefaultMetadataURIEnvVariable is the default environment variable used to hold the metadata endpoint URI.
-	DefaultMetadataURIEnvVariable = "ECS_CONTAINER_METADATA_URI"
+	// DefaultMetadataURIv3EnvVariable is the default environment variable used to hold the metadata endpoint URI v3.
+	DefaultMetadataURIv3EnvVariable = "ECS_CONTAINER_METADATA_URI"
+	// DefaultMetadataURIv4EnvVariable is the default environment variable used to hold the metadata endpoint URI v4.
+	DefaultMetadataURIv4EnvVariable = "ECS_CONTAINER_METADATA_URI_V4"
 
-	// Metadata v3 API paths
+	// Metadata v3 and v4 API paths. They're the same.
 	taskMetadataPath         = "/task"
 	taskMetadataWithTagsPath = "/taskWithTags"
 )
 
-// Client represents a client for a metadata v3 API endpoint.
+// Client represents a client for a metadata v3 or v4 API endpoint.
 type Client struct {
 	agentURL string
 }
 
-// NewClient creates a new client for the specified metadata v3 API endpoint.
+// NewClient creates a new client for the specified metadata v3 or v4 API endpoint.
```

Comment:
although I see the value of having a shared client for v3 and v4 (no code duplication). I think it's important that the client knows the api version it uses, it would be helpful when logging errors to print the accurate version of the api instead of `v3 or v4`

Reply:
perhaps we can pass it here in the constructor and set it as a propriety?

Reply:
Done in: https://github.com/DataDog/datadog-agent/pull/10727/commits/06fc659a00eeff102e617637170ec31d7483fc30